SDWebImage/SDWebImageManager.m

```diff
@@ -282,29 +282,33 @@ - (void)callDownloadProcessForOperation:(nonnull SDWebImageCombinedOperation *)o
-                            if (cacheSerializer) {
-                                cacheData = [cacheSerializer cacheDataWithImage:transformedImage  originalData:(imageWasTransformed ? nil : downloadedData) imageURL:url];
-                            } else {
-                                cacheData = (imageWasTransformed ? nil : downloadedData);
+                        @autoreleasepool {
+                            UIImage *transformedImage = [transformer transformedImageWithImage:downloadedImage forKey:key];
+                            if (transformedImage && finished) {
+                                NSString *transformerKey = [transformer transformerKey];
+                                NSString *cacheKey = SDTransformedKeyForKey(key, transformerKey);
+                                BOOL imageWasTransformed = ![transformedImage isEqual:downloadedImage];
+                                NSData *cacheData;
+                                // pass nil if the image was transformed, so we can recalculate the data from the image
+                                if (cacheSerializer) {
+                                    cacheData = [cacheSerializer cacheDataWithImage:transformedImage  originalData:(imageWasTransformed ? nil : downloadedData) imageURL:url];
+                                } else {
+                                    cacheData = (imageWasTransformed ? nil : downloadedData);
+                                }
+                                [self.imageCache storeImage:transformedImage imageData:cacheData forKey:cacheKey cacheType:storeCacheType completion:nil];
                             }
-                            [self.imageCache storeImage:transformedImage imageData:cacheData forKey:cacheKey cacheType:storeCacheType completion:nil];
+                            
+                            [self callCompletionBlockForOperation:strongOperation completion:completedBlock image:transformedImage data:downloadedData error:nil cacheType:SDImageCacheTypeNone finished:finished url:url];
                         }
-                        
-                        [self callCompletionBlockForOperation:strongOperation completion:completedBlock image:transformedImage data:downloadedData error:nil cacheType:SDImageCacheTypeNone finished:finished url:url];
                     });
                 } else {
                     if (downloadedImage && finished) {
                         if (cacheSerializer) {
                             dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
-                                NSData *cacheData = [cacheSerializer cacheDataWithImage:downloadedImage originalData:downloadedData imageURL:url];
-                                [self.imageCache storeImage:downloadedImage imageData:cacheData forKey:key cacheType:storeCacheType completion:nil];
+                                @autoreleasepool {
```

Comment:
@zhongwuzw I see. My previous comments meaning does not means we don't use autoreleasing pool. I means for some small logic block, espacially this one that may not contains logic to produce much autoreleasing object. It seems don't need to add one.

But anyway, brainlessly adding all block with autoreleasing pool is OK as well. Since it only contains temporary memory usage.

Reply:
@dreampiggy Emm, this maybe also needed, `cacheData` maybe has big size and it's autorelease object. So release timely is also useful IMO.